app/src/main/java/com/codesoom/assignment/application/AuthenticationService.java

```diff
@@ -0,0 +1,70 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.domain.User;
+import com.codesoom.assignment.domain.UserRepository;
+import com.codesoom.assignment.errors.UserAuthenticationFailedException;
+import com.codesoom.assignment.errors.UserEmailNotExistException;
+import com.codesoom.assignment.utils.JwtUtil;
+import org.springframework.stereotype.Service;
+
+/**
+ * 유저 인증에 대한 로직을 담당.
+ */
+@Service
+public class AuthenticationService {
+    private final JwtUtil jwtUtil;
+    private final UserRepository userRepository;
+
+    public AuthenticationService(JwtUtil jwtUtil, UserRepository userRepository) {
+        this.jwtUtil = jwtUtil;
+        this.userRepository = userRepository;
+    }
+
+    /**
+     * 주어진 유저 id를 부호화하여 토큰을 반환합니다.
+     *
+     * @param userId 유저 id
+     * @return 토큰
+     */
+    public String encode(Long userId) {
+        return jwtUtil.encode(userId);
+    }
+
+    /**
+     * 주어진 토큰을 해독하여 유저 id를 반환합니다.
+     *
+     * @param token 토큰
+     * @return 유저 id
+     */
+    public Long decode(String token) {
+        return jwtUtil.decode(token);
+    }
+
+    /**
+     * 주어진 유저 정보로 유저 인증 후 인증 토큰을 반환합니다.
```

Comment:
메소드 시그니처를 잘 읽어보세요. 다음과 같이 수정하면 어떨까요?
```suggestion
     * 주어진 유저 정보로 생성한 인증 토큰을 반환합니다.
```

Reply:
메소드 시그니처와는 동떨어진 주석을 작성하고 있었네요 ..
그리고 생각해보니 리턴값이 있는 메소드의 경우에는 JavaDoc Summary에 리턴값을 중심으로 설명하면 표현력이 더 좋아지는 것 같네요 !

Reply:
그렇습니다. Java 기본 라이브러리의 JavaDoc을 찾아서 열심히 읽어 보시기를 권하고 싶어요. 공식 JavaDoc에서는 리턴값을 굉장히 중요하게 여깁니다.

Reply:
조언 감사합니다 꾸준히 읽을수 있도록 습관을 들여야 겠어요 !